### PR TITLE
Add openClass and closeClass attributes

### LINKED ISSUE #219: Animate open and close modal
Hi,

is there a way to add class when the modal is open
and remove a class when the modal is close ?

I think it might be better to change 
`.css('display', 'block');` by  `.addClass(openClass)` Line 304
`.css('display', 'none');` by  `removeClass(openClass)` Line 329

What do you think about that ?

### PR DESCRIPTION
**Pull request by nlarche**
Hello Guy, 

I tried to do something, tell me If it's good to you and if you want me to do thing in another way
closes #219 

regards,

Nicolas

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -560,6 +561,14 @@ This option is to clear the search input when an item is selected. You need to s
 <input ion-autocomplete type="text" class="ion-autocomplete" autocomplete="off" ng-model="model" clear-on-select="true" />
 ```
 
+### Override open and close class
```
Comment (guylabs) [inline README.md]: Please use the following documentation (just some minor changes):

>### Open and close CSS class
>
>By default two CSS classes are used to display and hide the modal, namely the `ion-autocomplete-open` and the `ion-autocomplete-close` CSS class. These are used to show and hide the modal. When you need to override these classes, you can define the following two properties with your CSS class.
>
>```javascript
><input ion-autocomplete type="text" class="ion-autocomplete" open-class="my-open-class" close-class="my->close-class" />
>```

File: src/ion-autocomplete.css [hunk 2]
```diff
@@ -5,7 +5,7 @@
     bottom: 0;
     left: 0;
     z-index: 20;
-    display: none;
+    /*display: none;*/
```
Comment (guylabs) [inline src/ion-autocomplete.css]: Please delete it right away as commented out code is something you don't need ;).

File: src/ion-autocomplete.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ angular.module('ion-autocomplete', []).directive('ionAutocomplete', [
                 selectedItemsLabel: '@',
                 templateUrl: '@',
                 itemValueKey: '@',
-                itemViewValueKey: '@'
+                itemViewValueKey: '@'
```
Comment (guylabs) [inline src/ion-autocomplete.js]: Please remove the spaces you added at the end here.

File: src/ion-autocomplete.js [hunk 4]
```diff
@@ -477,7 +483,7 @@ angular.module('ion-autocomplete', []).directive('ionAutocomplete', [
                     // set the model value of the model
                     ngModelController.$parsers.push(function (viewValue) {
                         return ionAutocompleteController.getItemValue(viewValue, ionAutocompleteController.itemValueKey);
-                    });
+                    });
```
Comment (guylabs) [inline src/ion-autocomplete.js]: Please remove the spaces you added at the end here.

File: test/e2e/ion-autocomplete.single-select.e2e.html [hunk 5]
```diff
@@ -8,47 +8,60 @@
     <title>ion-autocomplete end to end test page</title>
 
     <link rel="stylesheet" href="../../bower_components/ionic/css/ionic.css">
-    <link rel="stylesheet" type="text/css" href="../../src/ion-autocomplete.css"/>
+    <link rel="stylesheet" type="text/css" href="../../src/ion-autocomplete.css" />
```
Comment (guylabs) [inline test/e2e/ion-autocomplete.single-select.e2e.html]: Please revert the formatting as this adds unneeded changes to the pull request.

File: test/e2e/ion-autocomplete.single-select.e2e.html [hunk 6]
```diff
@@ -8,47 +8,60 @@
...
 
     <script src="../../bower_components/ionic/js/ionic.bundle.js"></script>
     <script src="../../src/ion-autocomplete.js"></script>
     <script type="text/javascript">
         var app = angular.module('ion-autocomplete-sample', ['ionic', 'ion-autocomplete'])
-                .controller('IonAutocompleteController', function ($scope) {
```
Comment (guylabs) [inline test/e2e/ion-autocomplete.single-select.e2e.html]: Same here: Please revert the formatting as this adds unneeded changes to the pull request. (for the whole block from line 17 to 42)

File: test/e2e/ion-autocomplete.single-select.e2e.html [hunk 7]
```diff
@@ -8,47 +8,60 @@
...
+        }          
+
+    </style>
 </head>
 
 <body ng-controller="IonAutocompleteController">
-<ion-view title="ion-autocomplete sample">
-    <ion-content>
-        <form>
+    <ion-view title="ion-autocomplete sample">
```
Comment (guylabs) [inline test/e2e/ion-autocomplete.single-select.e2e.html]: Please revert the formatting as this adds unneeded changes to the pull request. Please check the rest of the file as there are more formatting changes like this.

**Comment by nlarche** [comment]
I made all changes and push dist files. it should be good.

**Comment by guylabs** [comment]
@nlarche Did you run the local e2e tests with protractor? Looks like the master is now failing: https://travis-ci.org/guylabs/ion-autocomplete/builds/204997944 Could you maybe have a look at it and check if you are able to reproduce it locally?

The strange thing is that here in the pull request it said the build was passing and the coverage increased. I am able to have a look at it on Monday evening if you cannot check it.

Thanks and regards,

Guy

**Comment by nlarche** [comment]
@guylabs yes it's weird, I ran all the test in local. I can't check until Monday morning.

**Comment by guylabs** [comment]
@nlarche I just fixed it as the open/close classes for the e2e tests did not work properly. I changed them an now all the tests work properly. Can you maybe check it on Monday such that I can release `0.4.0` in the evening? That would be great! 

Thanks a lot and have a good weekend.

**Comment by nlarche** [comment]
Ok, it looks good to me.

**Comment by guylabs** [comment]
Hey @nlarche, `0.4.0` is released. Thanks again for your contirbution!

Regards

Guy

**Pull request merged**